rest/route/pod.go

```diff
@@ -80,3 +89,92 @@ func (h *podAgentCedarConfig) Run(ctx context.Context) gimlet.Responder {
 	}
 	return gimlet.NewJSONResponse(data)
 }
+
+////////////////////////////////////////////////
+//
+// PUT /rest/v2/pods/{pod_id}
+
+type podIDPutHandler struct {
+	podID string
+	body  []byte
+	sc    data.Connector
+}
+
+func makePutPod(sc data.Connector) gimlet.RouteHandler {
+	return &podIDPutHandler{
+		sc: sc,
+	}
+}
+
+func (h *podIDPutHandler) Factory() gimlet.RouteHandler {
+	return &podIDPutHandler{
+		sc: h.sc,
+	}
+}
+
+// Parse fetches the podID and JSON payload from the HTTP request.
+func (h *podIDPutHandler) Parse(ctx context.Context, r *http.Request) error {
+	h.podID = gimlet.GetVars(r)["pod_id"]
+
+	body := util.NewRequestReader(r)
+	defer body.Close()
+	b, err := ioutil.ReadAll(body)
+	if err != nil {
+		return errors.Wrap(err, "Argument read error")
+	}
+	h.body = b
+
+	return nil
+}
+
+// Run creates a new resource based on the Request-URI and JSON payload.
+func (h *podIDPutHandler) Run(ctx context.Context) gimlet.Responder {
+	apiPod := &model.APICreatePod{
+		Name: utility.ToStringPtr(h.podID),
+	}
+	if err := json.Unmarshal(h.body, apiPod); err != nil {
+		return gimlet.MakeJSONInternalErrorResponder(errors.Wrap(err, "API error while unmarshalling JSON"))
+	}
+
+	newPod, respErr := validateCreatePod(ctx, apiPod, h.podID, true)
+	if respErr != nil {
+		return respErr
+	}
+
+	responder := gimlet.NewJSONResponse(struct{}{})
+	if err := responder.SetStatus(http.StatusCreated); err != nil {
+		return gimlet.MakeJSONInternalErrorResponder(errors.Wrapf(err, "Cannot set HTTP status code to %d", http.StatusCreated))
+	}
+	if err := h.sc.CreatePod(newPod); err != nil {
+		return gimlet.MakeJSONErrorResponder(errors.Wrapf(err, "Database error for insert() pod with pod id '%s", h.podID))
+	}
+
+	return responder
+}
+
+////////////////////////////////////////////////////////////////////////
+
+func validateCreatePod(ctx context.Context, apiPod *model.APICreatePod, resourceID string, isNewPod bool) (*pod.Pod, gimlet.Responder) {
```

Comment:
I would move this validation to the data connector since it has to do the REST -> DB conversion.